src/components/modules/theme-switch.tsx

```diff
@@ -3,18 +3,32 @@
 import { useTheme } from 'next-themes';
 
 import { MoonIcon, SunIcon } from '@radix-ui/react-icons';
-import { IconButton } from '@radix-ui/themes';
+import { IconButton, Spinner } from '@radix-ui/themes';
+import { useEffect, useState } from 'react';
 
 const ThemeSwitch = () => {
-  const { theme, setTheme } = useTheme();
+  const { resolvedTheme, setTheme } = useTheme();
+  const [mounted, setMounted] = useState(false)
+
+  useEffect(() => {
+    setMounted(true)
+  }, [])
 
   const onClick = () => {
-    setTheme(theme === 'dark' ? 'light' : 'dark');
+    setTheme(resolvedTheme === 'dark' ? 'light' : 'dark');
   };
 
+  if (!mounted) {
+    return (
+      <IconButton>
+        <Spinner />
+      </IconButton>
+    );
+  }
```

Comment:
```suggestion
  if (!resolvedTheme) {
    return (
      <IconButton>
        <Spinner />
      </IconButton>
    );
  }
```
We don't need a secondary state in this component to rerender this component. Instead, we can test if resolvedTheme is undefined, and infer which theme we're on based on that (since it takes some time to load when the theme should come from system).

Reply:
That's what I have done at the beginning but it fire some hydration error because resolvedTheme is always undefined first.
So I took the next theme doc example for this case.